examples/asr/speech_to_label_vad.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+# Preparing the dataset
+[TODO]
+Use the `process_vad_data.py` script under <NEMO_ROOT>/scripts in order to prepare the dataset.
+
+```sh
+python <NEMO_ROOT>/scripts/process_vad.py \
+    --data_root=<absolute path to where the data should be stored> \
+    --data_version=<either 1 or 2, indicating version of the dataset> \
+    --class_split=<either "all" or "sub", indicates whether all 30/35 classes should be used, or the 10+2 split should be used> \
+    --rebalance \
+    --log
+```
+
+# Train to convergence
+```sh
+python speech_to_label_vad.py \
+    # (Optional: --config-path=<path to dir of configs> --config-name=<name of config without .yaml>) \
+    model.train_ds.manifest_filepath="<path to train manifest>" \
+    model.validation_ds.manifest_filepath=["<path to val manifest>","<path to test manifest>"] \
+    trainer.gpus=2 \
+    trainer.distributed_backend="ddp" \
+    trainer.max_epochs=200 \
+    exp_manager.create_wandb_logger=True \
+    exp_manager.wandb_logger_kwargs.name="MatchboxNet-3x1x64-vad" \
+    exp_manager.wandb_logger_kwargs.project="MatchboxNet-vad" \
+    +trainer.precision=16 \
+    +trainer.amp_level=O1  # needed if using PyTorch < 1.6
+```
+"""
+import pytorch_lightning as pl
+
+from nemo.collections.asr.models import EncDecClassificationModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+@hydra_runner(config_path="conf", config_name="matchboxnet_3x1x64_vad")
+def main(cfg):
```

Comment:
Overall, other than the header above, this file is identical to speech_to_label.py. @blisc Do you think we should merge this file with speech_to_text.py and just add the header portion to switch the config?

Reply:
Yes, we might have different infer/eval script for different prediction methods and metrics. But for the training process, it's identical except the header.

Reply:
Ok let's merge the train script with above. Just append the docstring for vad and add two sections 1) Speech Commands 2) Voice activity detection with the respective commands underneath them.

Reply:
Good idea. `(config_path="conf", config_name="matchboxnet_3x1x64_vad")`  Do we want to keep it or make the config file mandatory? I think the second one is general?

Reply:
Keep the default as what it is inside speech to text.py. Add a line to the docs about how to switch that for vad config